packages/uniforms/__tests__/JSONSchemaBridge.js

```diff
@@ -0,0 +1,354 @@
+                        },
+                        required: ['type']
+                    }
+                ]
+            }
+        },
+        required: ['dateOfBirth']
+    };
+
+    const bridge = new JSONSchemaBridge(schema);
+
+    describe('#check()', () => {
+        it('works correctly with schema', () => {
+            expect(JSONSchemaBridge.check(schema)).toBeTruthy();
+        });
+
+        it('works correctly without schema', () => {
+            expect(JSONSchemaBridge.check()).not.toBeTruthy();
+        });
+    });
+
+    describe('#getError', () => {
+        it('works without error', () => {
+            expect(bridge.getError('age')).not.toBeTruthy();
+        });
+
+        it('works with invalid error', () => {
+            expect(bridge.getError('age', {})).not.toBeTruthy();
+            expect(bridge.getError('age', {invalid: true})).not.toBeTruthy();
+        });
+
+        it('works with correct error', () => {
+            expect(bridge.getError('age', {details: [{dataPath: '.age'}]})).toEqual({dataPath: '.age'});
+            expect(bridge.getError('age', {details: [{dataPath: '.field'}]})).not.toBeTruthy();
+        });
+    });
+
+    describe('#getErrorMessage', () => {
+        it('works without error', () => {
+            expect(bridge.getErrorMessage('age')).not.toBeTruthy();
+        });
+
+        it('works with invalid error', () => {
+            expect(bridge.getErrorMessage('age', {})).not.toBeTruthy();
+            expect(bridge.getErrorMessage('age', {invalid: true})).not.toBeTruthy();
+        });
+
+        it('works with correct error', () => {
+            expect(bridge.getErrorMessage('age', {details: [{dataPath: '.age', message: 'Zing!'}]})).toBe('Zing!');
+            expect(
+                bridge.getErrorMessage('age', {details: [{dataPath: '.field', message: 'Ignore!'}]})
+            ).not.toBeTruthy();
+        });
+    });
+
+    describe('#getErrorMessages', () => {
+        it('works without error', () => {
+            expect(bridge.getErrorMessages()).toEqual([]);
+        });
+
+        it('works with other errors', () => {
+            expect(bridge.getErrorMessages('correct')).toEqual(['correct']);
+            expect(bridge.getErrorMessages(999999999)).toEqual([999999999]);
+        });
+
+        it('works with Error', () => {
+            expect(bridge.getErrorMessages(new Error('correct'))).toEqual(['correct']);
+        });
+
+        it('works with ValidationError', () => {
+            expect(bridge.getErrorMessages({details: [{dataPath: '.age', message: 'Zing!'}]})).toEqual(['Zing!']);
+            expect(bridge.getErrorMessages({details: [{dataPath: '.field', message: 'Ignore!'}]})).toEqual(['Ignore!']);
+        });
+    });
+
+    describe('#getField', () => {
+        it('returns correct definition (flat)', () => {
+            expect(bridge.getField('age')).toEqual(expect.objectContaining({type: 'integer'}));
```

Comment:
Here (and all the way below): why aren't these tests exact (i.e. `toEqual(...)`) but using `arrayContaining`/`objectContaining`?